components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -686,30 +370,28 @@ void AIChatTabHelper::MakeAPIRequestWithConversationHistoryUpdate(
     question_part = turn.text;
   }
 
-  std::string prompt;
-  if (UsesLlama2PromptTemplate(ai_chat::features::kAIModelName.Get())) {
-    prompt = BuildLlama2Prompt(question_part);
-  } else {
-    prompt = BuildClaudePrompt(question_part, is_suggested_question);
-  }
-
-  if (turn.visibility != ConversationTurnVisibility::HIDDEN) {
-    AddToConversationHistory(turn);
-  }
+  // Suggested questions were based on only the initial prompt (with content),
+  // so no need to submit all conversation history when they are used.
+  std::vector<mojom::ConversationTurn> history =
+      is_suggested_question ? std::vector<mojom::ConversationTurn>()
+                            : chat_history_;
 
-  DCHECK(ai_chat_api_);
   auto data_received_callback = base::BindRepeating(
-      &AIChatTabHelper::OnAPIStreamDataReceived, weak_ptr_factory_.GetWeakPtr(),
-      current_navigation_id_);
+      &AIChatTabHelper::OnEngineCompletionDataReceived,
+      weak_ptr_factory_.GetWeakPtr(), current_navigation_id_);
```

Comment:
on a second though, the `current_navigation_id` will be incremented and both cells will be independently updated at the same time, correct?

Reply:
The only reason we have and use `current_navigation_id_` is to cancel any async operations after the page has navigated. We can therefore pretend that `current_navigation_id_` is some kind of "Conversation ID". If `current_navigation_id_` changes before the async callback is called, then we ignore the callback since it means it's not for the current conversation. This is code that hasn't changed in this PR.